tests/cross/src/test/scala/tests/pc/CompletionArgSuite.scala

```diff
@@ -699,4 +707,252 @@ class CompletionArgSuite extends BaseCompletionSuite {
+       |  def m(inn : Int, uuu: Option[Int]): Int = ???
+       |  def m(inn : Int, aaa: Int): Int = ???
+       |  def k: Int = m(inn = 1, a@@)
+       |}
+       |""".stripMargin,
+    """|aaa = : Int
+       |assert(assertion: Boolean): Unit
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "overloaded-generic",
+    """|object Main {
+       |  val h = 3
+       |  val l : List[Int] = List(1,2,3)
+       |  def m[T](inn : List[T], yy: Int, aaa: Int, abb: Option[Int]): Int = ???
+       |  def m[T](inn : List[T], yy: Int, aaa: Int, abb: Int): Int = ???
+       |  def k: Int = m(yy = 3, inn = l, a@@)
+       |}
+       |""".stripMargin,
+    """|aaa = : Int
+       |abb = : Int
+       |abb = : Option[Int]
+       |aaa = h : Int
+       |abb = h : Int
+       |""".stripMargin,
+    compat = Map(
+      "3" ->
+        """|aaa = : Int
+           |aaa = h : Int
+           |abb = : Option[Int]
+           |abb = : Int
+           |abb = h : Int
+           |""".stripMargin
+    ),
+    topLines = Some(5),
+  )
+
+  check(
+    "overloaded-methods",
+    """|class A() {
+       |  def m(anInt : Int): Int = ???
+       |  def m(aString : String): String = ???
+       |}
+       |object O {
+       |  def m(aaa: Int): Int = ???
+       |  val k = new A().m(a@@)
+       |}
+       |""".stripMargin,
+    """|aString = : String
+       |anInt = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  // type of `myInstance` is resolved to `null` for Scala 2
+  check(
+    "overloaded-methods2".tag(IgnoreScala2),
+    """|class A() {
+       |  def m(anInt : Int): Int = ???
+       |  def m(aString : String): String = ???
+       |  private def m(aBoolean: Boolean): Boolean = ???
+       |}
+       |object O {
+       |  def m(aaa: Int): Int = ???
+       |  val myInstance = new A()
+       |  val k = myInstance.m(a@@)
+       |}
+       |""".stripMargin,
+    """|aString = : String
+       |anInt = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "overloaded-select",
+    """|package a.b {
+       |  object A {
+       |    def m(anInt : Int): Int = ???
+       |    def m(aString : String): String = ???
+       |  }
+       |}
+       |object O {
+       |  def m(aaa: Int): Int = ???
+       |  val k = a.b.A.m(a@@)
+       |}
+       |""".stripMargin,
+    """|aString = : String
+       |anInt = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "overloaded-in-a-class",
+    """|trait Planet
+       |case class Venus() extends Planet
+       |class Main[T <: Planet](t : T) {
+       |  def m(inn: Planet, abb: Option[Int]): Int = ???
+       |  def m(inn: Planet, aaa: Int): Int = ???
+       |  def k = m(t, a@@)
+       |}
+       |""".stripMargin,
+    """|aaa = : Int
+       |abb = : Option[Int]
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "overloaded-function-param".tag(IgnoreScala2),
```

Comment:
Why are these ignored for Scala 2? It's fine to show a wrong output and a follow up issue.

Reply:
You can't have overloaded methods for Scala 2 that start w/ the same arg list. In theory you can define it but it's impossible to call it.
<img width="957" alt="Screenshot 2023-07-05 at 10 50 15" src="https://github.com/scalameta/metals/assets/26606662/ce685444-f671-45db-b43b-733868a082b7">

**Side note**: You can do it if the first arg list will disambiguate. For this reason for Scala 2, we only check the first arg list when looking for overloaded calls.
<img width="513" alt="Screenshot 2023-07-05 at 10 53 28" src="https://github.com/scalameta/metals/assets/26606662/ece9acbe-aae1-4dcc-af2e-752b4784afad">

Reply:
Ach ok., could you just add a comment? Thanks for explaning!